File: core/src/com/utils/TileAndCases.java
package com.utils;

import com.badlogic.gdx.math.Matrix3;
import com.badlogic.gdx.math.Vector2;
import com.tiles.Case;
import com.tiles.Pawn;
import com.tiles.Queue;
import com.tiles.Tile;

import java.util.ArrayList;

public class TileAndCases {
    /*
    C'est sans doute de l'hérétisme mais au moins ici je peux gérer toutes les variables facilement
    Boum, pour une fois que de l'hérétisme paye
    En vrai je sais pas si c'est de l'hérétisme, mais on nous avait dit que les classes statiques c'était caca donc bon...
     */
    public static float tileSize = 600;
    public static float offset = 40 * tileSize / 600;
    public static float caseSize = (tileSize - 2 * offset) / 4;
    public static Vector2 origin = new Vector2();
    public static ArrayList<Tile> tileList;
    public static Case lastExploredCase;

    public static boolean isInPhaseB = false;

    public static Matrix3 newBase = new Matrix3(new float[]{tileSize, -caseSize, 0f, caseSize, tileSize, 0f, 0f, 0f, 1f});
    public static Matrix3 newBaseInvert = new Matrix3(newBase).inv();

    public static Queue queue=null;

    public static ArrayList<Pawn> pawnList = new ArrayList();


}

File: core/src/com/utils/TileAndCases.kt
package com.utils

import com.badlogic.gdx.math.Matrix3
import com.badlogic.gdx.math.Vector2
import com.screens.game.board.Case
import com.screens.game.board.Pawn
import com.screens.game.board.Queue
import com.screens.game.board.Tile

/**
 * Objet pour stocker différentes valeurs.
 */
val tileSize = 600f
val offset = 40 * tileSize / 600
val caseSize = (tileSize - 2 * offset) / 4
val origin = Vector2()
val portalList: Map<Color, MutableList<Case>> = Color.values().associateWith { mutableListOf() }
var lastExploredCase: Case? = null
var isInPhaseB = false
var numberWeaponsRetrieved = 0
var numberPawnsOut = 0
var newBase = Matrix3(floatArrayOf(tileSize, -caseSize, 0f, caseSize, tileSize, 0f, 0f, 0f, 1f))
var newBaseInvert: Matrix3 = Matrix3(newBase).inv()
var queue: Queue? = null

val tileList = mutableListOf<Tile>()
val pawnList = mutableListOf<Pawn>()
